client/blocks/jitm/index.jsx

```diff
@@ -19,25 +22,32 @@ export const JITM = ( {
 	featureClass,
 	id,
 	message,
-} ) => ( hasJitm && currentSite ) && (
-	<Banner
-		callToAction={ callToAction }
-		title={ message }
-		description={ description }
-		disableHref
-		dismissPreferenceName={ featureClass + '123' }
-		dismissTemporary={ true }
-		event={ `jitm_nudge_click_${ id }` }
-		href={ `https://jetpack.com/redirect/?source=jitm-${ id }&site=${ currentSite.domain }` }
-	/>
-);
+	onDismiss,
+} ) =>
+	hasJitm &&
+	currentSite && (
+		<Banner
+			callToAction={ callToAction }
+			title={ message }
+			description={ description }
+			disableHref
+			dismissPreferenceName={ featureClass + '123' }
+			dismissTemporary={ true }
+			onDismiss={ onDismiss( currentSite.ID, id, featureClass ) }
+			event={ `jitm_nudge_click_${ id }` }
+			href={ `https://jetpack.com/redirect/?source=jitm-${ id }&site=${ currentSite.domain }` }
+		/>
+	);
+
+const mapStateToProps = state => ( {
+	currentSite: getSelectedSite( state ),
+	hasJitm: !! getTopJITM( state ),
+	...getTopJITM( state ),
+} );
 
-const mapStateToProps = ( state ) => (
-	{
-		currentSite: getSelectedSite( state ),
-		hasJitm: !! getTopJITM( state ),
-		...getTopJITM( state ),
-	}
-);
+const mapDispatchToProps = dispatch => ( {
+	onDismiss: ( siteId, id, featureClass ) => () =>
+		dismissJetpackJITM( dispatch, siteId, id, featureClass ),
```

Comment:
this is probably missing a call to `dispatch()`…

Reply:
@dmsnell, what do you mean?

Reply:
oh, I actually read it wrong, but I'm glad you asked - we definitely don't want to be passing `dispatch` in to an action creator. _this_ place in the code is the appropriate place to dispatch an action - we want the action creators to be able to create their action objects without affecting the system, leaving the actual effects (triggered by `dispatch`) to be left at the edges, which in this case are on a button click.

```js
onDismiss: ( … ) => dispatch( dismissJetpackJITM( siteId, id, featureClass ) );
```